pkg/devcontainer/setup/lifecyclehooks.go

```diff
@@ -96,9 +97,13 @@ func run(commands []types.LifecycleHook, user, dir string, remoteEnv map[string]
 
 		for k, c := range cmd {
 			log.Infof("Run command %s: %s...", k, strings.Join(c, " "))
+			currentUser, err := user.Current()
+			if err != nil {
+				log.Fatalf("Error fetching the current user: %v", err)
+			}
 			args := []string{}
-			if user != "root" {
-				args = append(args, "su", user, "-c", command.Quote(c))
+			if remoteUser != currentUser.Username {
```

Comment:
oh we are comparing usernames, I thought we compared User structs. Is it possible to update the function signature to pass the actual remoteUser struct to `run` and compare UID instead of username

Reply:
I think this won't be easy because the `remoteUser` value here comes from the `GetRemoteUser` function. `GetRemoteUser` returns the user name and not the id, partially also because it relies on  devcontainer.json's `remoteUser` - which specifies the user name, and not the user id.